spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,260 @@
+        self.setLayout(installer_layout)
+
+        # Signals
+        self.sig_download_progress.connect(
+            self._installation_widget.update_installation_progress)
+        self.sig_installation_status.connect(
+            self._installation_widget.update_installation_status)
+        self.sig_installation_status.connect(
+            self.finished_installation)
+
+        self._installation_widget.ok_button.clicked.connect(
+            self.close_installer)
+        self._installation_widget.cancel_button.clicked.connect(
+            self.cancel_install)
+
+        # Show integration widget
+        self.setup()
+
+    def setup(self, installation=False):
+        """Setup visibility of widgets."""
+        self._installation_widget.setVisible(True)
+        self.adjustSize()
+
+    def cancel_install(self):
+        """Cancel the installation in progress."""
+        reply = QMessageBox.critical(
+            self._parent, 'Spyder',
+            _('Do you really want to cancel the Spyder update installation?'),
+            QMessageBox.Yes, QMessageBox.No)
+        if reply == QMessageBox.Yes:
+            self.cancelled = True
+            self.cancel_thread_install_update()
+            self.setup()
+            self.accept()
+            return True
+        return False
+
+    def continue_install(self):
+        """
+        Continue the installation in progress by downloading the installer.
+        """
+        reply = QMessageBox(icon=QMessageBox.Question,
+                            text=_('Do you want to download and'
+                                   ' install the latest version of'
+                                   ' Spyder?<br>'),
+                            parent=self._parent)
+        reply.setWindowTitle("Spyder")
+        reply.setAttribute(Qt.WA_ShowWithoutActivating)
+        reply.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+        reply.exec_()
+        if reply.result() == QMessageBox.Yes:
+            self.start_installation_update()
+        else:
+            self._change_update_installation_status(status=PENDING)
+
+    def finished_installation(self, status):
+        """Handle finished installation."""
+        if status == FINISHED or status == PENDING:
+            self.setup()
+            self.accept()
+
+    def close_installer(self):
+        """Close the installation dialog."""
+        if (self.status == FINISHED
+                or self.status == CANCELLED):
+            self.setup()
+            self.accept()
+        else:
+            self.hide()
+
+    def reject(self):
+        """Reimplemented Qt method."""
+        on_installation_widget = self._installation_widget.isVisible()
+        if on_installation_widget:
+            self.close_installer()
+        else:
+            super(UpdateInstallerDialog, self).reject()
+
+    def _change_update_installation_status(self, status=NO_STATUS):
+        """Set the installation status."""
+        self.status = status
```

Comment:
Lets add some debug logging here. For that, you will need to import the `logging` module and then at the begining of the file create a `logger` instance (similar setup at the one done in the `status.py` file). After that, we can add here something like:

```python
logger.debug(f"Installation status: {status}"))`
```